tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/EagerSession.java

```diff
@@ -306,6 +308,15 @@ public void checkInput(Op input) {
     }
   }
 
+  @Override
+  public synchronized Scope baseScope() {
+    if(baseScope == null){
```

Comment:
Turns out creating the scope in initialization causes issues, since the native handles haven't been set yet.  I tried a field and a `{ }` initializer, neither worked.

Reply:
On which native handles an instance of `Scope` depends on?

Reply:
And did you tried to initialize it in the constructor instead?

Reply:
It needs it to get the ops for the name import.  And I probably can use the constructor, but it's a little dangerous if we change initialization at some point (since I'm passing a partially initialized `this` to Scope).

Reply:
Sorry I’m still not seeing it, can you point to the line of code showing that instantiating a `Scope` requires a native handle?

Also while your concern about using an incomplete `this` sounds hazardous, I think I still prefer this to the current lazy synchronized initialization.

Reply:
> Sorry I’m still not seeing it, can you point to the line of code showing that instantiating a Scope requires a native handle?

`Scope()` calls `NameScope()`, which calls `withUsedFrom`, which calls `Graph.operations()` (https://github.com/tensorflow/java/blob/bc9096a09a42169dce599a2a019ab99f02853de5/tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/op/NameScope.java#L61)

> Also while your concern about using an incomplete this sounds hazardous, I think I still prefer this to the current lazy synchronized initialization.

Agreed.